File: diffx-python/tests/docs_examples/performance_benchmarks_examples.py
import pytest
import diffx_python as diffx

def test_performance_benchmarks_example_1():
    result = diffx.benchmark_diff("content1", "content2")
    assert result is not None

def test_performance_benchmarks_example_2():
    result = diffx.benchmark_diff("content1", "content2", output_format="json")
    assert result is not None

def test_performance_benchmarks_example_3():
    result = diffx.benchmark_diff("content1", "content2", iterations=100)
    assert result is not None

def test_performance_benchmarks_example_4():
    result = diffx.benchmark_diff("content1", "content2", memory_profile=True)
    assert result is not None

def test_performance_benchmarks_example_5():
    result = diffx.benchmark_diff("content1", "content2", cpu_profile=True)
    assert result is not None

def test_performance_benchmarks_example_6():
    result = diffx.benchmark_diff("content1", "content2", warmup=10)
    assert result is not None

def test_performance_benchmarks_example_7():
    result = diffx.benchmark_diff("content1", "content2", no_color=True)
    assert result is not None

def test_performance_benchmarks_example_8():
    result = diffx.benchmark_diff("content1", "content2", algorithm="myers")
    assert result is not None

def test_performance_benchmarks_example_9():
    result = diffx.benchmark_diff("content1", "content2", threads=4)
    assert result is not None

def test_performance_benchmarks_example_10():
    result = diffx.benchmark_diff("content1", "content2", save_results="benchmark_results.json")
    assert result is not None

File: diffx-core/tests/performance_benchmarks_examples.rs
use diffx_core::{DiffEngine, BenchmarkConfig};

#[test]
fn performance_benchmarks_example_1() {
    let engine = DiffEngine::new();
    let config = BenchmarkConfig::default();
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_2() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.output_format = "json".to_string();
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_3() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.iterations = 100;
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_4() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.memory_profile = true;
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_5() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.cpu_profile = true;
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_6() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.warmup_iterations = 10;
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_7() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.no_color = true;
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_8() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.algorithm = "myers".to_string();
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_9() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.threads = 4;
    let _result = engine.benchmark_diff("content1", "content2", &config);
}

#[test]
fn performance_benchmarks_example_10() {
    let engine = DiffEngine::new();
    let mut config = BenchmarkConfig::default();
    config.save_results = Some("benchmark_results.json".to_string());
    let _result = engine.benchmark_diff("content1", "content2", &config);
}